proxmox/resource_vm_qemu.go

```diff
@@ -223,15 +321,11 @@ func resourceVmQemu() *schema.Resource {
 				},
 			},
 			"disk": &schema.Schema{
-				Type:          schema.TypeSet,
+				Type:          schema.TypeList,
 				Optional:      true,
 				ConflictsWith: []string{"disk_gb", "storage", "storage_type"},
 				Elem: &schema.Resource{
 					Schema: map[string]*schema.Schema{
-						"id": &schema.Schema{
```

Comment:
From my understanding of the code and PVE API, I think removing this will have unintended side-effects. For example, what if, in my Terraform file, I have 3 disks - 1, 2, & 3. With this change, if I remove disk 2, it ends up mount the old disk 2 volume in the disk 3 (now 2) slot.